src/halmos/sevm.py

```diff
@@ -1138,23 +1144,23 @@ def init(cls, ex: Exec, addr: Any, slot: int, keys: tuple) -> None:
         assert_address(addr)
         num_keys = len(keys)
         size_keys = cls.bitsize(keys)
-        if slot not in ex.storage[addr]:
-            ex.storage[addr][slot] = {}
-        if num_keys not in ex.storage[addr][slot]:
-            ex.storage[addr][slot][num_keys] = {}
-        if size_keys not in ex.storage[addr][slot][num_keys]:
+        if slot not in ex.storage[addr].mapping:
+            ex.storage[addr].mapping[slot] = {}
+        if num_keys not in ex.storage[addr].mapping[slot]:
+            ex.storage[addr].mapping[slot][num_keys] = {}
+        if size_keys not in ex.storage[addr].mapping[slot][num_keys]:
             if size_keys == 0:
-                if ex.symbolic:
+                if ex.storage[addr].symbolic:
                     label = f"storage_{id_str(addr)}_{slot}_{num_keys}_{size_keys}_00"
-                    ex.storage[addr][slot][num_keys][size_keys] = BitVec(
+                    ex.storage[addr].mapping[slot][num_keys][size_keys] = BitVec(
                         label, BitVecSort256
                     )
                 else:
-                    ex.storage[addr][slot][num_keys][size_keys] = ZERO
+                    ex.storage[addr].mapping[slot][num_keys][size_keys] = ZERO
             else:
                 # do not use z3 const array `K(BitVecSort(size_keys), ZERO)` when not ex.symbolic
                 # instead use normal smt array, and generate emptyness axiom; see load()
-                ex.storage[addr][slot][num_keys][size_keys] = cls.empty(
+                ex.storage[addr].mapping[slot][num_keys][size_keys] = cls.empty(
```

Comment:
take this as a soft suggestion and please double check if it is equivalent. I just applied a couple of my heuristics:
- `setdefault` instead of `if x not in y: y[x] = ...`
- avoid repeated dict lookups, factor them out in their own variable (e.g. `storage = ex.storage[addr]`)
- reduce nesting with early exits when applicable (e.g. `if size_keys in slot_mapping[num_keys]: return`)
- move variable definitions close to their point of use

Gives the following:

```py
    @classmethod
    def init(cls, ex: Exec, addr: Any, slot: int, keys: tuple) -> None:
        assert_address(addr)
        storage = ex.storage[addr]
        storage.mapping.setdefault(slot, {})

        num_keys = len(keys)
        slot_mapping = storage.mapping[slot]
        slot_mapping.setdefault(num_keys, {})

        size_keys = cls.bitsize(keys)
        if size_keys in slot_mapping[num_keys]:
            return

        if size_keys != 0:
            # do not use z3 const array `K(BitVecSort(size_keys), ZERO)` when not ex.symbolic
            # instead use normal smt array, and generate emptyness axiom; see load()
            slot_mapping[num_keys][size_keys] = cls.empty(addr, slot, keys)
            return

        # size_keys == 0
        if storage.symbolic:
            label = f"storage_{id_str(addr)}_{slot}_{num_keys}_{size_keys}_00"
            slot_mapping[num_keys][size_keys] = BitVec(label, BitVecSort256)
        else:
            slot_mapping[num_keys][size_keys] = ZERO
```

Reply:
taking it a couple steps further:

- factoring out `slot_mapping[num_keys]` 
- using the return value of setdefault)
- turn if else blocks into a single expression

becomes:

```python
    @classmethod
    def init(cls, ex: Exec, addr: Any, slot: int, keys: tuple) -> None:
        assert_address(addr)
        storage = ex.storage[addr]
        slot_mapping = storage.mapping.setdefault(slot, {})

        num_keys = len(keys)
        slot_mapping_num_keys = slot_mapping.setdefault(num_keys, {})

        size_keys = cls.bitsize(keys)
        if size_keys in slot_mapping_num_keys:
            return

        if size_keys != 0:
            # do not use z3 const array `K(BitVecSort(size_keys), ZERO)` when not ex.symbolic
            # instead use normal smt array, and generate emptyness axiom; see load()
            slot_mapping_num_keys[size_keys] = cls.empty(addr, slot, keys)
            return

        # size_keys == 0
        slot_mapping_num_keys[size_keys] = (
            mk_uint256(f"storage_{id_str(addr)}_{slot}_{num_keys}_{size_keys}_00")
            if storage.symbolic
            else ZERO
        )
```

Reply:
could be improved with a better name for `slot_mapping_num_keys`, but I don't have a good intuition for what it should be

Reply:
done in #356